File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int left = 0, right = 0, ans =0;
        unordered_map<char,int> m;
        while(right < s.size()){
            if(m.count(s[right])){
                left = max(left, m[s[right]] + 1);
            }
            m[s[right]] = right;
            ans = max(ans, right - left + 1);
            right++;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-1.java
class Solution {
    private boolean isPossible(int[] piles,double num,int h){
        int count = 0;
        for(int pile:piles){
            count += Math.ceil(pile/num);
            if(count > h)return false;
        }
        return true;
    }
    public int minEatingSpeed(int[] piles, int h) {
        int maxi = 0;
        for(int pile:piles){
            maxi = Math.max(pile,maxi);
        }
        int left = 1, right = maxi;
        int ans = -1;
        while(left <= right){
            int mid = (left + right) / 2;
            if(isPossible(piles,mid,h)){
                System.out.println(mid);
                right = mid-1;
                ans = mid;
            }
            else{
                left = mid+1;
            }
        }
        return ans;
    }
}
